### PR TITLE
Scrape SIS for Major Requirements

### PR DESCRIPTION
**Pull request by DC00**

### CODE REVIEW

File: lib/sis/sis.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+driver.find_element(:name => 'DERIVED_SSS_SCL_SSS_MORE_ACADEMICS').send_key('w')
+driver.find_element(:name => 'DERIVED_SSS_SCL_SSS_GO_1').click
+wait = Selenium::WebDriver::Wait.new(:timeout => 10)
+wait.until { driver.find_element(:name => 'DERIVED_SAAWHIF_SSS_CREATE_NEW').displayed? }
+driver.find_element(:name => 'DERIVED_SAAWHIF_SSS_CREATE_NEW').click
+wait = Selenium::WebDriver::Wait.new(:timeout => 10)
+wait.until { driver.find_element(:name => 'DERIVED_SAAWHIF_ACAD_PROG$0').displayed? }
+driver.find_element(:name => 'DERIVED_SAAWHIF_ACAD_PROG$0').send_key('Arts & Sciences')
+driver.find_element(:name => 'PLAN$0').send_key('Economics (BA)')
+driver.execute_script("submitAction_win0(document.win0,'DERIVED_SAAWHIF_SSR_PB_SUBMIT');")
```
Comment (DC00) [inline lib/sis/sis.rb]: s/o Alan

**Comment by kienan** [comment]
This is a good example of something we should probably move to another repository!

**Comment by DC00** [comment]
@kienan The wiki/wiki/wiki repository?

**Comment by kienan** [comment]
lol -- but actually, this seems like it's own tool. what do you think about creating a new repository for it?

**Comment by DC00** [comment]
It's a good idea. I'll make one

**Comment by aw3as** [comment]
the conclusion of this script needs access to the rails models - thus, needs to be in this repository

File: lib/sis/sis.rb [hunk 2]
```diff
@@ -51,13 +57,15 @@
...
+puts 'generating report...'
 # wait for report to be finished
 wait.until { driver.find_element(:id => 'win0divDERIVED_SAA_DPR_GROUPBOX1$2').displayed? }
 driver.find_element(:name => 'DERIVED_SAA_DPR_SSS_EXPAND_ALL').click
 # wait for expansion to finish
 wait.until { driver.find_element(:id => 'WAIT_win0').displayed? }
 wait.until { !driver.find_element(:id => 'WAIT_win0').displayed? }
 
+puts 'parsing information...'
 major_element = driver.find_element(:id => 'win0divDERIVED_SAA_DPR_GROUPBOX1$2')
```
Comment (DC00) [inline lib/sis/sis.rb]: `win0divDERIVED_SAA_DPR_GROUPBOX1$2 ` is the id for each of the purple table headers. I think we previously assumed there would be a constant number of purples, but Econ has 4 and I know CS only has 3. Need to find a way to make sure it's the major and not the general area requirements

**Comment by DC00** [comment]
screen shot 2016-09-08 at 5 47 13 pm

General outline

**Comment by emumoo** [comment]
This is so cool

**Comment by DC00** [comment]
Having a hard time parsing the tables holding the classes within each subcategory

File: lib/sis/sis.rb [hunk 3]
```diff
@@ -67,25 +75,40 @@
...
+
+puts major[:text]
+major[:categories].each do |category|
+  puts "\t#{category[:name]}"
+  puts "\t#{category[:text]}"
+  category[:subcategories].each do |subcategory|
+    puts "\t\t#{subcategory[:name]}"
+    puts "\t\t#{subcategory[:text]}"
+  end
+end
```
Comment (kienan) [inline lib/sis/sis.rb]: Add a newline here plz

**Pull request closed without merging**